slash-commands/moderation/timeout.js

```diff
@@ -0,0 +1,181 @@
+const Discord = require('discord.js');
+const dateformat = require('dateformat');
+const ms = require('ms');
+const {verifyReasonLength} = require('../../helpers/stringHelpers');
+const {SlashCommandBuilder} = require('@discordjs/builders');
+const {promisePool} = require('../../config/db');
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('timeout')
+    .setDefaultPermission(false)
+    .setDescription('Timeout a user')
+    .addUserOption((option) =>
+      option
+        .setName('user')
+        .setDescription('The user to timeout')
+        .setRequired(true)
+    )
+    .addStringOption((option) =>
+      option
+        .setName('duration')
+        .setDescription(
+          'How long the user should be timed out as <number><s|m|h|d|w>'
+        )
+        .setRequired(true)
+    )
+    .addStringOption((option) =>
+      option
+        .setName('reason')
+        .setDescription('The reason for the timeout')
+        .setRequired(true)
+    ),
+
+  async execute(interaction) {
+    let toTimeout;
+    interaction.guild.members
+      .fetch(await interaction.options.getUser('user'))
+      .then((result) => (toTimeout = result));
+    const duration = await interaction.options.getString('duration');
+    const reason = await interaction.options.getString('reason');
+
+    if (canTimeout(interaction, toTimeout, duration, reason)) {
+      try {
+        await timeoutUser(interaction, toTimeout, duration, reason);
+      } catch (e) {
+        console.log(e);
+        interaction.reply(
+          'Something went wrong while trying to timeout the user.'
+        );
+      }
+      await auditLogTimeout(interaction, toTimeout, duration, reason);
+      await recordTimeoutInDB(interaction, toTimeout, duration, reason);
+      try {
+        await dmTheUser(interaction, toTimeout, duration, reason);
+      } catch (e) {
+        console.log(e);
+        interaction.reply(`${e.name}: ${e.message}`);
+      }
+    }
+  },
+};
+
+// Validates data.
```

Comment:
I think we can use the helper functions isServerStaff and sendNoTargetStaffReply in helpers/validation instead of canTimeout. The same logic is used in many of the commands so it makes sense to have it in one place. Those functions also disallow use of the command on the Code Counselor role.

Reply:
Will be addressed in a future issue and PR for all commands based on Fede's input in Discord.